spec/classes/agent_upgrade_spec.rb

```diff
@@ -14,26 +14,33 @@
               })
             end
 
-            [{}, {:service_names => []}].each do |params|
-              context "agent_upgrade class without any parameters" do
-                let(:params) { params }
+            if Puppet.version < "3.8.0"
+              it { expect { is_expected.to contain_package('agent_upgrade') }.to raise_error(Puppet::Error, /upgrading requires Puppet 3.8/) }
+            elsif Puppet.version >= "4.0.0"
+              # TODO: How to test this warning and lack of changes?
+              it { is_expected.to contain_package('agent_upgrade') }
```

Comment:
Not sure how I want to handle this. It might be better to have Puppet 4+ behavior match 3.8, with an additional warning. Right now the module skips everything for Puppet 4+.